api/tests/test_audit_log.py

```diff
@@ -0,0 +1,176 @@
+    qs = AuditLogEntry.objects.filter(
+        message__audit_event__target__path=url,
+        message__audit_event__operation=Operation.UPDATE.value,
+    )
+    assert qs.count() == 1
+    audit_log_entry = qs.first()
+
+    # Verify the audit log entry details
+    assert audit_log_entry.message["audit_event"]["target"]["object_ids"] == [
+        str(delivery_log.pk)
+    ]
+    assert audit_log_entry.message["audit_event"]["target"]["path"] == url
+    assert audit_log_entry.message["audit_event"]["operation"] == Operation.UPDATE.value
+
+
+def test_deleting_delivery_log_writes_delete_log(admin_user):
+    """
+    Test that deleting a DeliveryLog object writes a delete log.
+    """
+    client = Client()
+    client.force_login(admin_user)
+
+    delivery_log = DeliveryLogFactory()
+
+    # Get the URL for deleting the DeliveryLog object
+    url = reverse("admin:api_deliverylog_delete", args=[delivery_log.pk])
+
+    # Send a POST request to delete the object
+    response = client.post(url, {"post": "yes"})
+    assert response.status_code == 302  # Redirect after successful deletion
+
+    # Assert that an AuditLogEntry was created for the deletion
+    qs = AuditLogEntry.objects.filter(
+        message__audit_event__target__path=url,
+        message__audit_event__operation=Operation.DELETE.value,
+    )
+    assert qs.count() == 1
+    audit_log_entry = qs.first()
+
+    # Verify the audit log entry details
+    assert audit_log_entry.message["audit_event"]["target"]["object_ids"] == [
+        str(delivery_log.pk)
+    ]
+    assert audit_log_entry.message["audit_event"]["target"]["path"] == url
+    assert audit_log_entry.message["audit_event"]["operation"] == Operation.DELETE.value
+    assert DeliveryLog.objects.count() == 0
+
+
+def test_bulk_deleting_delivery_logs_writes_delete_log(admin_user):
+    """
+    Test that bulk deleting DeliveryLog objects writes a delete log.
+    """
+    client = Client()
+    client.force_login(admin_user)
+
+    COUNT = 3
+
+    delivery_logs = DeliveryLogFactory.create_batch(COUNT)
+    object_ids = [str(log.pk) for log in delivery_logs]
+
+    # Get the URL for the changelist view
+    url = reverse("admin:api_deliverylog_changelist")
+
+    # Send a POST request to bulk delete the objects
+    data = {
+        "action": "delete_selected",
+        "_selected_action": object_ids,
+        "post": "yes",
+    }
+    response = client.post(url, data)
+    assert response.status_code == 302  # Redirect after successful deletion
+
+    # Assert that an AuditLogEntry was created for the bulk deletion
+    qs = AuditLogEntry.objects.filter(
+        message__audit_event__target__path=url,
+        message__audit_event__operation=Operation.DELETE.value,
+    )
+    assert qs.count() == 1
+    audit_log_entry = qs.first()
+
+    # Verify the audit log entry details
+    assert len(audit_log_entry.message["audit_event"]["target"]["object_ids"]) == COUNT
```

Comment:
Adding the testing of the removed IDs:
```suggestion
    assert len(audit_log_entry.message["audit_event"]["target"]["object_ids"]) == COUNT
    assert sorted(
        audit_log_entry.message["audit_event"]["target"]["object_ids"]
    ) == sorted(object_ids)    
```